### PR TITLE
test(frontend): add sendBtc tests

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

As agreed, before adding changes to a function, I'm covering existing functionality with tests. In this case, it's about `sendBtc` of `btc-send.services`.

### CODE REVIEW

File: src/frontend/src/tests/btc/services/btc-send.services.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import * as btcSendService from '$btc/services/btc-send.services';
+import { type SendBtcParams } from '$btc/services/btc-send.services';
+import { convertNumberToSatoshis } from '$btc/utils/btc-send.utils';
+import * as backendAPI from '$lib/api/backend.api';
+import * as signerAPI from '$lib/api/signer.api';
+import { ProgressStepsSendBtc } from '$lib/enums/progress-steps';
+import { mapToSignerBitcoinNetwork } from '$lib/utils/network.utils';
+import * as walletUtils from '$lib/utils/wallet.utils';
```
Comment (peterpeterparker) [inline src/frontend/src/tests/btc/services/btc-send.services.spec.ts:8]: Is it possible to not mock the walletUtils or that too much complexity?

File: src/frontend/src/tests/btc/services/btc-send.services.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		network: 'mainnet',
+		source: 'address',
+		destination: 'address',
+		identity: mockIdentity,
+		amount: 10
+	} as SendBtcParams;
+	const txid = 'txid;';
+
+	beforeEach(() => {
+		mockPage.reset();
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/tests/btc/services/btc-send.services.spec.ts:27]: True, removed

File: src/frontend/src/tests/btc/services/btc-send.services.spec.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import * as btcSendService from '$btc/services/btc-send.services';
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/tests/btc/services/btc-send.services.spec.ts:1]: Just a leftover, changed it

File: src/frontend/src/tests/btc/services/btc-send.services.spec.ts (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/tests/btc/services/btc-send.services.spec.ts:8]: Actually, it's not required to mock it, my idea was to make sure that it has been called once. If you think it doesn't bring any benefits, I can just remove it, no strong opinion here.

**Comment by DenysKarmazynDFINITY** [comment]
@peterpeterparker added additional "should throw" tests, please take a look.

File: src/frontend/src/tests/btc/services/btc-send.services.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/tests/btc/services/btc-send.services.spec.ts:7]: Generally speaking, in my opinion, the less the mocks, the better so if not too hard to not mock, yes sure if the test can be asserted without mocking, go for it when it comes to me

**Pull request merged**